File: src/main/java/io/curiousoft/ijudi/ordermanagement/validator/ValidDeliveryInfo.java
package io.curiousoft.ijudi.ordermanagement.validator;

import javax.validation.Constraint;
import javax.validation.Payload;

import java.lang.annotation.Documented;
import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.FIELD;
import static java.lang.annotation.ElementType.TYPE;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Target({ FIELD, TYPE })
@Retention(RUNTIME)
@Constraint(validatedBy = ShippingValidator.class)
@Documented
public @interface ValidDeliveryInfo {

    String message() default "Delivery address or building or unit number not correct.";

    Class<?>[] groups() default { };

    Class<? extends Payload>[] payload() default { };
}

File: izinga-commons/src/main/kotlin/io/curiousoft/izinga/commons/validator/ValidDeliveryInfo.kt
package io.curiousoft.izinga.commons.validator

import java.lang.annotation.Documented
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy
import javax.validation.*
import kotlin.reflect.KClass

@Target(AnnotationTarget.FIELD, AnnotationTarget.ANNOTATION_CLASS, AnnotationTarget.CLASS)
@Retention(RetentionPolicy.RUNTIME)
@Constraint(validatedBy = [ShippingValidator::class])
@Documented
annotation class ValidDeliveryInfo(
    val message: String = "Delivery address or building or unit number not correct.",
    val groups: Array<KClass<*>> = [],
    val payload: Array<KClass<out Payload>> = []
)